BENCH-70-compile and build answerking app using github-actions

JoeCSykes opened the pull request.

JoeCSykes committed:
Create ci.yml

JoeCSykes committed:
put python version as string

PietroConvalleAD commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,27 @@
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
is it the latest LTS (long time support) or just latest release?

JoeCSykes replied:
I think it currently runs 18.04 LTS but will be switching to 20.04 LTS on Nov 30th (see https://github.blog/changelog/2020-10-29-github-actions-ubuntu-latest-workflows-will-use-ubuntu-20-04/). Seen as there is 22.04 LTS I think it runs the latest ubuntu LTS that they have tested against github workflows. As such I think were ok with just putting `ubuntu-latest`

PietroConvalleAD replied:
that article is from 2020, https://github.com/actions/runner-images#available-images looks like it's the 22.04 LTS 👍

PietroConvalleAD commented:
`/home/runner/work/AnswerKing-Python/AnswerKing-Python/answerking_python does not contain any element
Error: Process completed with exit code 1.`

we have this error from some package during install but the install is actually alright. we have to find a workaround.

PietroConvalleAD commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,27 @@
+        poetry install
+    - name: Run Tests
+      run: |
+        python manage.py test
```
`poetry run python manage.py test`?

JoeCSykes committed:
change poetry install to poetry update

PietroConvalleAD commented on .github/workflows/ci.yml:
```diff
@@ -21,7 +21,7 @@ jobs:
         python -m pip install --upgrade pip
         pip install poetry
-        poetry install
+        poetry update
```
why this change?

PietroConvalleAD replied:
oh, well, I see fixes the error that we were receiving, not sure why and how but ok cool

JoeCSykes committed:
use poetry to run tests

PietroConvalleAD commented:
mysql set up is missing.

JoeCSykes committed:
added set up for SQL DB and created .env file

JoeCSykes commented:
Need to change DBPWD to equal a GitHub secret but currently don't have permissions to create a secret on this project. I have messaged Oleksii to see whether he wants to give me permissions or whether he wants me to do it another way. I feel like using secrets is the safest way to do this.

JoeCSykes committed:
unsafe password for DBPWD to test works

I have changed the DBPWD variable to a generic string and set the root password to this as well to test if it works. I plan to change it to a github secret variable once I have permissions to do this.

JoeCSykes committed:
set test as separate job that needs build

JoeCSykes committed:
corrected test job

PietroConvalleAD commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,58 @@
+  
+  
+  test:
+    runs-on: ubuntu-latest
```
My guess is that it's failing because you are redefining the environment and hence you don't have the dependencies installed

JoeCSykes committed:
set up mysql server in services

JoeCSykes committed:
sorted indent problem

JoeCSykes committed:
added Django secret key to env

JoeCSykes committed:
changed env names to match django

JoeCSykes committed:
changed database name

PietroConvalleAD commented:
needs linting and typing check

JoeCSykes committed:
added tests for linting and type checking

PietroConvalleAD commented on .github/workflows/ci.yml:
```diff
@@ -46,6 +46,9 @@ jobs:
     - name: Run Tests     
       run: |
         poetry run python manage.py test
+        poetry run black . --line-length=79
```
I think we want to run `poetry run black . --line-length=79 --check` since we are just looking if the files are formatted properly

JoeCSykes committed:
changed so black is checked rather than run

PietroConvalleAD reviewed (dismissed):
Looks good to me, the only think left is using GH secrets

JoeCSykes committed:
moved env variables so don't have to restate

abbas-khan8 commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,52 @@
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
```
Do we want to add main to this?

JoeCSykes replied:
can do if people want?

JoeCSykes replied:
added main

PietroConvalleAD commented:
Needs rebase with development now

JoeCSykes committed:
changed so GitHub workflow runs on updates to main

JoeCSykes committed:
added DJANGO_SETTINGS_MODULE key to github env

JoeCSykes committed:
Test for test fail on github workflow

JoeCSykes committed:
Merge branch 'BENCH-70-compile-and-build-answerking-app-using-github-actions' of https://github.com/AnswerConsulting/AnswerKing-Python into BENCH-70-compile-and-build-answerking-app-using-github-actions

JoeCSykes committed:
Reverted test to original. Github reported error fine.

JoeCSykes committed:
Test to trigger black error

JoeCSykes committed:
separated out tests into different steps

JoeCSykes committed:
corrected black error test

JoeCSykes committed:
Test pycodestyle fail

JoeCSykes committed:
revert changes for testing pycodestyle error

JoeCSykes committed:
change type of id to str to test pyright fail on github workflow

PietroConvalleAD commented on answerking_app/utils/mixins/CategoryItemMixins.py:
```diff
@@ -12,7 +12,7 @@
 class CategoryItemUpdateMixin:
     def update(
-        self, request: Request, cat_id: int, item_id: int, *args, **kwargs
+        self, request: Request, cat_id: str, item_id: int, *args, **kwargs
```
why?

PietroConvalleAD replied:
oh you are testing the fails, nvm

JoeCSykes committed:
corrected pyright type error
